src/Corax/IndexWriter.cs

```diff
@@ -599,15 +601,19 @@ public void Commit()
             }
         }
 
-        private unsafe void AddNewTerm(List<long> entries, CompactTree fieldTree, ReadOnlySpan<byte> termsSpan, Span<byte> tmpBuf)
+        private unsafe void AddNewTerm(List<long> entries, CompactTree fieldTree, ReadOnlySpan<byte> termsSpan, Span<byte> tmpBuf, CompactTree.EncodedKey encodedKey, bool validEncodedKey = true)
```

Comment:
Maybe make `encodedKey` nullable and get rid of `validEncodedKey`

Reply:
Rationale behind passing in the validEncodedKey: because encodedKey could be out of date (due to deletion) so we have to generate it before we add the term.  We cannot make EncodedKey nullable either because it is a read-only ref struct